### PR TITLE
MWPW-134751 Blog Block Report

### PR DESCRIPTION
**Pull request by Brandon32**
* Adding reporting tool to identify the types of and number of blocks.

Resolves: [MWPW-134705](https://jira.corp.adobe.com/browse/MWPW-134705)

### CODE REVIEW

File: block-report.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    }
+}
+
+async function runReport(project, site, indexUrl, cached = true) {
+    const blocks = {};
+    const variants = {};
+    const totals = { blocks: {}, variants: {} };
+    const entries = await loadIndex(project, indexUrl, cached);
+
+    for (const entry of entries) {
```
Comment (fullcolorcoder) [inline block-report.js:74]:
```suggestion
    for (let i = 0; i < entries.length; i++) {
        const entry = entries[i];
```

File: block-report.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+                variants[entry] = variants[entry] || {};
+                variants[entry][variant] = (variants[entry][variant] || 0) + 1;
+                totals.variants[variant] = (totals.variants[variant] || 0) + 1;
+            }
+            blocks[entry][blockName] = (blocks[entry][blockName] || 0) + 1;
+            totals.blocks[blockName] = (totals.blocks[blockName] || 0) + 1;
+        });
+
+
+        console.log(entry, blocks[entry]);
```
Comment (fullcolorcoder) [inline block-report.js:96]:
```suggestion
        console.log(entry, blocks[entry]);
        if (i < entries.length - 1) {
            await new Promise((resolve) => setTimeout(resolve, 500));
        }        
```

**Review by fullcolorcoder — APPROVED** [review]
This is awesome. I suggested a few lines to throttle the code, seen some comments around that recently. 500ms is what the Franklin team seems to be going with.

File: block-report.js (hunk 2 above)
Comment (Brandon32) [reply block-report.js:96]: Sounds good, however I'll add the delay in `fetchMarkdown` so to no slow down cached content.

**Review by meganthecoder — APPROVED** [review]

**Pull request merged**